### PR TITLE
Fix Pastel Veil mechanics

### PR DESCRIPTION
**Pull request by urkerab**
Makes Pastel Veil pass the tests from #6568

### CODE REVIEW

File: data/abilities.js [hunk 1]
```diff
@@ -2651,6 +2651,20 @@ let BattleAbilities = {
...
+				}
+			}
+		},
+		onUpdate(pokemon) {
+			if (['psn', 'tox'].includes(pokemon.status)) {
+				this.add('-activate', pokemon, 'ability: Pastel Veil');
+				pokemon.cureStatus();
+			}
+		},
 		onAllySwitchIn(pokemon) {
```
Comment (TheImmortal) [inline data/abilities.js:2668]: Is this necessary? I never read anything that mentioned it in @DaWoblefet's research.

Comment (DaWoblefet) [reply data/abilities.js:2668]: Not entirely sure what it's doing in particular, but if I'm reading this right, Pastel Veil should announce itself when blocking poison (except for switching in on Toxic Spikes), indicate its source, and indicate the target being cured or being made immune to poison.

Comment (TheImmortal) [reply data/abilities.js:2668]: It cures the ally's status when the ally switches in. Not the Pastel Veil user, they're already active.

Comment (DaWoblefet) [reply data/abilities.js:2668]: That behavior is correct too, it should cure an ally's poison on switchin if the other active Pokemon is Pastel Veil.

Comment (Zarel) [reply data/abilities.js:2668]: Isn't `onUpdate` enough to cover the others?

Comment (Zarel) [reply data/abilities.js:2668]: Oh, `onUpdate` only affects the user.

**Pull request merged**